### PR TITLE
Add security to linux guidance

### PR DESCRIPTION
**Pull request by shirhatti**

### CODE REVIEW

**Comment by shirhatti** [comment]
@Rick-Anderson Can you do an UE review?

File: aspnet/publishing/linuxproduction.rst [hunk 1]
```diff
@@ -174,3 +151,83 @@ You can redirect application logs (``STDOUT`` and ``STERR``) in the program sect
 
     tail -f /var/log/hellomvc.out.log
 
+Securing our application
+------------------------
+
+Enable ``AppArmor``
+~~~~~~~~~~~~~~~~~~~
+Linux Security Modules (LSM) is a framework that is part of the Linux kernel since Linux 2.6 that supports different implementations of security modules. ``AppArmor`` is a LSM that implements a Mandatory Access Control system which allows you to confine program to a limited set of resources. Ensure `AppArmor <https://wiki.ubuntu.com/AppArmor>`__ is enabled and properly configured.
```
Comment (Rick-Anderson) [inline aspnet/publishing/linuxproduction.rst]: >confine program to a limited

confine the program to a limited

File: aspnet/publishing/linuxproduction.rst [hunk 2]
```diff
@@ -174,3 +151,83 @@ You can redirect application logs (``STDOUT`` and ``STERR``) in the program sect
 
     tail -f /var/log/hellomvc.out.log
 
+Securing our application
```
Comment (Rick-Anderson) [inline aspnet/publishing/linuxproduction.rst]: Consider Securing the application or securing the app. Why our?

File: aspnet/publishing/linuxproduction.rst [hunk 3]
```diff
@@ -174,3 +151,83 @@ You can redirect application logs (``STDOUT`` and ``STERR``) in the program sect
...
+------------------------
+
+Enable ``AppArmor``
+~~~~~~~~~~~~~~~~~~~
+Linux Security Modules (LSM) is a framework that is part of the Linux kernel since Linux 2.6 that supports different implementations of security modules. ``AppArmor`` is a LSM that implements a Mandatory Access Control system which allows you to confine program to a limited set of resources. Ensure `AppArmor <https://wiki.ubuntu.com/AppArmor>`__ is enabled and properly configured.
+
+Configuring our firewall
+~~~~~~~~~~~~~~~~~~~~~~~~
+Close off all external ports that are not in use. Uncomplicated firewall (ufw) provides a frontend for ``iptables`` by providing a command-line interface for configuring the firewall. Ensure that ``ufw`` is installed and available and configured allow traffic on any ports you may need.
```
Comment (Rick-Anderson) [inline aspnet/publishing/linuxproduction.rst]: and configured *to* allow

Comment (Rick-Anderson) [inline aspnet/publishing/linuxproduction.rst]: >installed and available and configured
can't we just say configured and enabled?

maybe:
Verify that ``ufw`` is configured to allow traffic on ports you need.

File: aspnet/publishing/linuxproduction.rst [hunk 4]
```diff
@@ -174,3 +151,83 @@ You can redirect application logs (``STDOUT`` and ``STERR``) in the program sect
...
+    sudo apt-get install ufw
+    sudo ufw enable
+
+    sudo ufw allow 80/tcp
+    sudo ufw allow 443/tcp
+
+Securing Nginx
+~~~~~~~~~~~~~~
+
+The default distribution of Nginx isn't configured securely with SSL enabled. To ensure we have all the security features we want, we will build from source.
```
Comment (Rick-Anderson) [inline aspnet/publishing/linuxproduction.rst]: The default distribution of Nginx  doesn't enable SSL. To enable all the security features we require, we will ..

File: aspnet/publishing/linuxproduction.rst [hunk 5]
```diff
@@ -174,3 +151,83 @@ You can redirect application logs (``STDOUT`` and ``STERR``) in the program sect
...
+.. code-block:: c
+
+    static char ngx_http_server_string[] = "Server: Your Web Server" CRLF;
+    static char ngx_http_server_full_string[] = "Server: Your Web Server" CRLF;
+
+
+Configure the options and build
+```````````````````````````````
+
+The PCRE library is required for regular expressions in the location directive for the ngx_http_rewrite_module. The http_ssl_module adds HTTPS protocol support.
```
Comment (Rick-Anderson) [inline aspnet/publishing/linuxproduction.rst]: something like
The PCRE library is required for regular expressions. regular expressions are used in the ``location`` directive for the ngx_http_rewrite_module.

File: aspnet/publishing/linuxproduction.rst [hunk 6]
```diff
@@ -174,3 +151,83 @@ You can redirect application logs (``STDOUT`` and ``STERR``) in the program sect
...
+
+You should also considering using a web application firewall like `ModSecurity` to further harden your application.
```
Comment (Rick-Anderson) [inline aspnet/publishing/linuxproduction.rst]: ~~You should also~~

File: aspnet/publishing/linuxproduction.rst [hunk 7]
```diff
@@ -174,3 +151,83 @@ You can redirect application logs (``STDOUT`` and ``STERR``) in the program sect
...
+    ./configure
+    --with-pcre=../pcre-8.38
+    --with-zlib=../zlib-1.2.8
+    --with-http_ssl_module
+    --with-stream
+    --with-mail=dynamic
+
+Configure SSL
+`````````````
+Configure your server to listen to HTTPS traffic on port ``443`` by specifying a valid certificate issued by a trusted Certificate Authority (CA). Harden your security by employing some of the practices suggested below like choosing a stronger cipher and redirecting all traffic over HTTP to HTTPS. Additionally, adding an ``HTTP Strict-Transport-Security`` (HSTS) header ensures all subsequent requests made by the client are over HTTPS only. Do not add the Strict-Transport-Security header or chose an appropriate ``max-age`` if you plan to disable SSL in the future.
```
Comment (Rick-Anderson) [inline aspnet/publishing/linuxproduction.rst]: ~~Additionally, ~~

Comment (Rick-Anderson) [inline aspnet/publishing/linuxproduction.rst]: Consider making this a bullet list.

File: aspnet/publishing/linuxproduction.rst (hunk 2 above)
Comment (shirhatti) [inline aspnet/publishing/linuxproduction.rst]: I'll leave it as our for consistency now. I'll do another pass later

**Comment by Rick-Anderson** [comment]
@shirhatti ping

**Comment by shirhatti** [comment]
I'll take a pass at your feedback later today 👍

**Comment by shirhatti** [comment]
@Rick-Anderson Just addressed your feedback

**Pull request merged**